halo2_proofs/src/plonk/permutation/prover.rs

```diff
@@ -159,12 +160,17 @@ impl Argument {
                 z.push(tmp);
             }
             let mut z = domain.lagrange_from_vec(z);
-            // Set blinding factors
-            for z in &mut z[params.n() as usize - blinding_factors..] {
-                *z = Scheme::Scalar::random(&mut rng);
+            if ZK {
+                // Set blinding factors
+                for z in &mut z[params.n() as usize - blinding_factors..] {
+                    *z = Scheme::Scalar::random(&mut rng);
+                }
+                // Set new last_z
+                last_z = z[params.n() as usize - (blinding_factors + 1)];
+            } else {
+                // Set new last_z
+                last_z = *z.last().unwrap() * modified_values.last().unwrap();
             }
-            // Set new last_z
-            last_z = z[params.n() as usize - (blinding_factors + 1)];
 
             let blind = Blind(Scheme::Scalar::random(&mut rng));
```

Comment:
Should we set the blind as zero here for non-zk setting?

Reply:
I think even we set it zero, the underlying commitment scheme still has lots of to do with commitment blinding factors, and I prefer to leave it untouched because I'm not super familiar with IPA.

Reply:
Blinding factors in commitments are handled in commitment scheme side. For example KZG variants just ignores them while IPA applies. If they are also not useful with non ZK IPA `ProverIPA` should know about it. I think generating random blinds with non ZK setting won't effect performance much and since there are too much of them I also prefer not touching them at constraint system side to reduce the diff.

Reply:
That makes sense to me. I think the zero-knowledge is achieved by the the randomness padding with the hiding property of only leaking one evaluation. Thus I guess the blinding factor is only used in IPA so it can be set to zero by IPA itself.